commit 0e01256a314a6f432ab9826dc9b862e8159dbc48
Author: Christopher Faulet <cfaulet@haproxy.com>
Date:   Wed Sep 18 11:07:20 2019 +0200

    BUG/MINOR: mux-h2: Be sure to have a connection to unsubcribe
    
    When the mux is released, It must own the connection to unsubcribe.
    This patch must be backported to 2.0.
    
    (cherry picked from commit 21d849f52fc64c51e5abf5a8bd69f2aeff8b3125)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/src/mux_h2.c b/src/mux_h2.c
index 984a81bd..e6bfd03d 100644
--- a/src/mux_h2.c
+++ b/src/mux_h2.c
@@ -677,9 +677,9 @@ static void h2_release(struct h2c *h2c)
 		}
 		if (h2c->wait_event.tasklet)
 			tasklet_free(h2c->wait_event.tasklet);
-		if (h2c->wait_event.events != 0)
+		if (conn && h2c->wait_event.events != 0)
 			conn->xprt->unsubscribe(conn, conn->xprt_ctx, h2c->wait_event.events,
-			    &h2c->wait_event);
+						&h2c->wait_event);
 
 		pool_free(pool_head_h2c, h2c);
 	}
